### DISCUSSION TITLE
Use Guzzle delay option instead of blocking via usleep

### DISCUSSION THREAD

**Discussion by ostrolucky** [spatie/guzzle-rate-limiter-middleware#25] [Ideas]
Guzzle provides `delay` option https://docs.guzzlephp.org/en/stable/request-options.html#delay which can be used to delay sending of the request. CurlMultiHandler utilises it in non-blocking manner. It would be great if this middleware used it instead of calling `usleep()`

**Comment by freekmurze** [comment]
Feel free to send a PR for this. All current test should remain green.
